From c8fce8e6e918549c31838f263e6e257c7bbe5bdf Mon Sep 17 00:00:00 2001
From: Zhang Zekun <zhangzekun11@huawei.com>
Date: Mon, 14 Nov 2022 09:15:34 +0000
Subject: [PATCH 0062/1398] [Huawei] ACPI / APEI: Add config to isolate Notify
 all ras err

Offering: HULK
hulk inclusion
category: feature
bugzilla: 187974

--------------------------------------------------

Add CONFIG_ACPI_APEI_NOTIFY_ALL_RAS_ERR to isolate Notify all
ras err to driver, which was introduced in
924ceaed70a63e5eaaf39b78dc756e4a7ac3c250.

Signed-off-by: Zhang Zekun <zhangzekun11@huawei.com>
---
 arch/arm64/configs/openeuler_defconfig | 1 +
 drivers/acpi/apei/Kconfig              | 7 +++++++
 drivers/acpi/apei/ghes.c               | 6 +++++-
 3 files changed, 13 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/configs/openeuler_defconfig b/arch/arm64/configs/openeuler_defconfig
index 24300e2f76e2..3544ea64b79d 100644
--- a/arch/arm64/configs/openeuler_defconfig
+++ b/arch/arm64/configs/openeuler_defconfig
@@ -719,6 +719,7 @@ CONFIG_HAVE_ACPI_APEI=y
 CONFIG_ACPI_APEI=y
 CONFIG_ACPI_APEI_GHES=y
 CONFIG_ACPI_APEI_GHES_TS_CORE=y
+CONFIG_ACPI_APEI_GHES_NOTIFY_ALL_RAS_ERR=y
 CONFIG_ACPI_APEI_PCIEAER=y
 CONFIG_ACPI_APEI_SEA=y
 CONFIG_ACPI_APEI_MEMORY_FAILURE=y
diff --git a/drivers/acpi/apei/Kconfig b/drivers/acpi/apei/Kconfig
index f33c23062471..bf65cce12f61 100644
--- a/drivers/acpi/apei/Kconfig
+++ b/drivers/acpi/apei/Kconfig
@@ -40,6 +40,13 @@ config ACPI_APEI_GHES_TS_CORE
 	help
 	  Enable the support for ts core ras process for ascend.
 
+config ACPI_APEI_GHES_NOTIFY_ALL_RAS_ERR
+	bool "Notify all ras err to driver"
+	depends on ARM64 && ACPI_APEI_GHES
+	default n
+	help
+	  Deliver all types of error to driver.
+
 config ACPI_APEI_PCIEAER
 	bool "APEI PCIe AER logging/recovering support"
 	depends on ACPI_APEI && PCIEAER
diff --git a/drivers/acpi/apei/ghes.c b/drivers/acpi/apei/ghes.c
index 632c6584ceea..5ad7c74eff00 100644
--- a/drivers/acpi/apei/ghes.c
+++ b/drivers/acpi/apei/ghes.c
@@ -667,14 +667,18 @@ static bool ghes_do_proc(struct ghes *ghes,
 #endif
 		} else {
 			void *err = acpi_hest_get_payload(gdata);
-
+#ifndef CONFIG_ACPI_APEI_GHES_NOTIFY_ALL_RAS_ERR
+			ghes_defer_non_standard_event(gdata, sev);
+#endif
 			log_non_standard_event(sec_type, fru_id, fru_text,
 					       sec_sev, err,
 					       gdata->error_data_length);
 		}
 
+#ifdef CONFIG_ACPI_APEI_GHES_NOTIFY_ALL_RAS_ERR
 		/* Customization deliver all types error to driver. */
 		ghes_defer_non_standard_event(gdata, sev);
+#endif
 	}
 
 	return queued;
-- 
2.17.1

